### PR TITLE
Fix/Expanded Create VM with TF

### PR DESCRIPTION
**Pull request by aevans-ukcloud**
Can now create multiple VM's with either managed or unmanaged disks for either Windows or Linux

### CODE REVIEW

File: articles/azure/azs-how-create-vm-terraform.md [hunk 1]
```diff
@@ -84,33 +77,18 @@ Official [Variables Guide](https://www.terraform.io/intro/getting-started/variab
...
-    client_id       = "{applicationId}@{tenantDomain}"
-    client_secret   = "{applicationPassword}"
-    tenant_id       = "xxxxx-xxxx-xxxx-xxxx-xxxxxxxxxx"
-  }
-  ```
-
-#### Argument Reference
+#### Argument reference
 
 - `arm_endpoint` - The Azure Resource Manager API Endpoint for your Azure Stack instance. This will be **`https://management.{region}.{domain}`**.
```
[the remark this answers is not in the document]
Comment (BaileyLawson) [reply articles/azure/azs-how-create-vm-terraform.md]: Can we still reword this?: For UKCloud Region it is **`https://management.frn00006.azure.ukcloud.com`**.

File: articles/azure/azs-how-create-vm-terraform.md [hunk 2]
```diff
@@ -125,34 +103,65 @@ Official [Variables Guide](https://www.terraform.io/intro/getting-started/variab
...
-    name              = "myosdisk1"
-    caching           = "ReadWrite"
-    create_option     = "FromImage"
-    managed_disk_type = "Standard_LRS"
-  }'>A managed disk
-  </p>
-</form>
-
-### Create a VM with Public IP template - using **Environment Variables** option
+The examples that follow show how to create VMs using Terraform. The code changes depending on whether you are creating a VM with an unmanaged disk or a managed disk. Select the the type of disk you want to use to update the examples below with the appropriate code.
```
[the remark this answers is not in the document]
Comment (BaileyLawson) [reply articles/azure/azs-how-create-vm-terraform.md]: Still needs fixing

File: articles/azure/azs-how-create-vm-terraform.md [hunk 3]
```diff
@@ -93,7 +93,7 @@ The process of authentication can be handled in one of two ways, either as **Env
 - `arm_endpoint` - The Azure Resource Manager API Endpoint for your Azure Stack instance. This will be **`https://management.{region}.{domain}`**.
 
 > [!NOTE]
-> For UKCloud Region it is **`https://management.frn00006.azure.ukcloud.com`**.
+> For UKCloud the arm_endpoint is **`https://management.frn00006.azure.ukcloud.com`**.
```
Comment (BaileyLawson) [inline articles/azure/azs-how-create-vm-terraform.md]: Need a comma after "UKCloud"

File: articles/azure/azs-how-create-vm-terraform.md [hunk 4]
```diff
@@ -130,23 +130,23 @@ Official [Variables Guide](https://www.terraform.io/intro/getting-started/variab
 
 ## Creating a VM
 
-The examples that follow show how to create VMs using Terraform. The code changes depending on whether you are creating a VM with an unmanaged disk or a managed disk. Select the the type of disk you want to use to update the examples below with the appropriate code.
+The examples that follow show how to create VMs using Terraform. The code changes depending on whether you are creating a VM with an unmanaged or managed disk. You can select the the type of disk you want to use, which will then update the examples below with the appropriate code.
```
Comment (BaileyLawson) [inline articles/azure/azs-how-create-vm-terraform.md]: "the the"

File: articles/azure/azs-how-create-vm-terraform.md [hunk 5]
```diff
@@ -130,23 +130,23 @@ Official [Variables Guide](https://www.terraform.io/intro/getting-started/variab
...
...er 2012 Datacenter - Pay as you use</option><option value="MicrosoftWindowsServer/WindowsServer/2012-R2-Datacenter/latest">Windows Server 2012 R2 Datacenter - Pay as you use</option><option value="MicrosoftWindowsServer/WindowsServer/2016-Datacenter/latest">Windows Server 2016 Datacenter - Pay-as-you-use</option><option value="MicrosoftWindowsServer/WindowsServer/2016-Datacenter-Server-Core/latest">Windows Server 2016 Datacenter - Server Core - Pay as you use</option><option value="MicrosoftWindowsServer/WindowsServer/2016-Datacenter-with-Containers/latest">Windows Server 2016 Datacenter - with Containers - Pay as you use</option><option value="MicrosoftWindowsServer/WindowsServer/2019-Datacenter/latest">Windows Server 2019 Datacenter</option><option value="MicrosoftWindowsServer/WindowsServer/2019-Datacenter-with-Containers/latest">Windows Server 2019 Datacenter with Containers</option><option value="MicrosoftWindowsServer/WindowsServer/2019-Datacenter-Core/latest">Windows Server 2019 Datacenter Core</option><option value="MicrosoftWindowsServer/WindowsServer/2019-Datacenter-Core-with-Containers/latest">Windows Server 2019 Datacenter Core with Containers</option></select></form> |
```
Comment (BaileyLawson) [inline articles/azure/azs-how-create-vm-terraform.md]: System doesn't need to be capitalised

File: articles/azure/azs-how-create-vm-terraform.md [hunk 6]
```diff
@@ -159,7 +159,7 @@ The examples that follow show how to create VMs using Terraform. The code change
 ### Defining the required Azure Stack resources in the `main.tf` file
 
 > [!TIP]
-> Resource attributes such as "name" can also be hardcoded as opposed to through the use of variables, allowing for the use of `main.tf` without the external variable files.
+> Resource attributes such as "name" can be hardcoded as opposed to using variables from the external files for the declaration
```
Comment (BaileyLawson) [inline articles/azure/azs-how-create-vm-terraform.md]: Full stop

File: articles/azure/azs-how-create-vm-terraform.md [hunk 7]
```diff
@@ -125,34 +103,65 @@ Official [Variables Guide](https://www.terraform.io/intro/getting-started/variab
...
-    name              = "myosdisk1"
-    caching           = "ReadWrite"
-    create_option     = "FromImage"
-    managed_disk_type = "Standard_LRS"
-  }'>A managed disk
-  </p>
-</form>
-
-### Create a VM with Public IP template - using **Environment Variables** option
+The examples that follow show how to create VMs using Terraform. The code changes depending on whether you are creating a VM with an unmanaged or managed disk. You can select the the type of disk you want to use, which will then update the examples below with the appropriate code.
```
Comment (BaileyLawson) [inline articles/azure/azs-how-create-vm-terraform.md]: follow should be follows

File: articles/azure/azs-how-create-vm-terraform.md [hunk 8]
```diff
@@ -125,34 +103,65 @@ Official [Variables Guide](https://www.terraform.io/intro/getting-started/variab
...
+
+## Declare variables
 
+| Variable Name        | Variable Description                                                                                                        | Input                                                        |
+|----------------------|-----------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------|
+| arm_endpoint       | The Azure Resource Manager API Endpoint for your Azure Stack instance. This will be https://management.{region}.{domain}.   | <form oninput="result.value=arm_endpoint.value" id="arm_endpoint" style="display:inline;"><input type="text" id="arm_endpoint" name="arm_endpoint" style="display:inline;" placeholder="https://management.{region}.{domain}"/></form> |
```
Comment (BaileyLawson) [inline articles/azure/azs-how-create-vm-terraform.md]: Endpoint shouldn't be capitalised

File: articles/azure/azs-how-create-vm-terraform.md [hunk 9]
```diff
@@ -125,34 +103,65 @@ Official [Variables Guide](https://www.terraform.io/intro/getting-started/variab
...
+| subscription_id    | The ID of your Azure Stack Subscription.    | <form oninput="result.value=subscription_id.value" id="subscription_id" style="display:inline;"><input type="text" id="subscription_id" name="subscription_id" style="display:inline;" placeholder="xxxxx-xxxx-xxxx-xxxx-xxxxxxxxxx"/></form> |
```
Comment (BaileyLawson) [inline articles/azure/azs-how-create-vm-terraform.md]: Subscription shouldn't be capitalised

File: articles/azure/azs-how-create-vm-terraform.md [hunk 10]
```diff
@@ -125,34 +103,65 @@ Official [Variables Guide](https://www.terraform.io/intro/getting-started/variab
...
+| rg_tag   | An optional tag to help categorize the Resource Group, e.g. "Production"    | <form oninput="result.value=rg_tag.value" id="rg_tag" style="display:inline;"><input type="text" id="rg_tag" name="rg_tag" style="display:inline;" placeholder="Production"/></form>
```
Comment (BaileyLawson) [inline articles/azure/azs-how-create-vm-terraform.md]: Resource and group don't need to be capitalised

File: articles/azure/azs-how-create-vm-terraform.md [hunk 11]
```diff
@@ -368,96 +717,120 @@ resource "azurestack_virtual_machine" "test" {
...
 }
-</code></pre>
+</pre></code>
 
-### How to execute a plan - using **Environment Variables** option
+***
 
-> [!TIP]
-> Terraform by default scans your execution directory and looks for all `tf` files.
+### Assign values to the variables in the `terraform.tfvars` file through the table Found at the top of the document
```
Comment (BaileyLawson) [inline articles/azure/azs-how-create-vm-terraform.md]: Found shouldn't be capitalised

File: articles/azure/azs-how-create-vm-terraform.md (hunk 7 above)
Comment (aevans-ukcloud) [reply articles/azure/azs-how-create-vm-terraform.md]: I believe examples that follow is correct, would need be follows if it was "example"

Comment (BaileyLawson) [reply articles/azure/azs-how-create-vm-terraform.md]: Yep, my bad

**Pull request merged**